backend/inspirehep/matcher/utils.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from bs4 import BeautifulSoup
+from inspire_dojson.utils import strip_empty_values
+from inspire_utils.helpers import force_list
+
+
+def get_arxiv_eprint(beatutiful_soup_xml_obj):
+    try:
+        return beatutiful_soup_xml_obj.find("idno", attrs={"type": "arxiv"}).text
```

Comment:
use `xpath` with `parsel` instead of beautiful soup like we usually do, e.g. https://github.com/inspirehep/hepcrawl/blob/master/hepcrawl/parsers/jats.py? this will fail miserably if you have more than one arXiv reference. I would also use the same approach as there by having a class + properties with a `parse` method calling the builder on the properties.

Reply:
It's great acutally that we have `ReferenceBuilder`, I should have checked